### DISCUSSION TITLE
Possibility to use GET requests for GraphQL fetching

### DISCUSSION THREAD

**Discussion by cobrani** [wpengine/faustjs#981] [General]
I'm currently migrating my website from Frontity framework to Faustjs. So far, I really the framework. However, I have the need to deliver the results from GraphQL through a CDN for website performance. I know there are upcoming CDNs for this, but I'd rather use HTTP GET requests for the moment as it has also (limited) support in WP GraphQL plugin (see https://www.wpgraphql.com/docs/interacting-with-wpgraphql/). So far, I don't see a possibility to do that in Faustjs or GQty, but maybe you have an idea to accomplish that? Thank you

**Comment by theodesp** [comment]
@cobrani you can assign your own `queryFetcher` function when you use the `getClient` function:

https://faustjs.org/docs/next/reference/getClient#queryfetcher

This way you can use GET requests instead of POST requests, however you would need to encode them using the right format.

`http://localhost:8080/graphql?query={...}&variables={...}&operation=...`

**Reply by cobrani** [reply]
Thanks a lot for your answer @theodesp ! I tried to change the method in the meantime but I missed using the correct format for variables. I'll try again asap I have time and check how it works. Didn't expect that it's so easy to change the fetching - this is awesome!
As you changed the issue to discussion type, I might add some things I thought about (after my first try to change the fetching method). Maybe it helps people who consider using a CDN with faustjs:
- For CDN, it might be better to put the whole next.js behind CDN, so assets etc. can be also delivered through it
- I noticed (this was not clear to me at first), that all requests to GraphQL endpoint are coming from my server directly, there is no client request at all - so CDN just for the GraphQL endpoint might be too much and also not needed at all (pls correct me if I'm wrong)
- If I put the whole next js behind CDN, static site generation must be considered regarding revalidate time, `Cache-Control` header with `stale-while-revalidate` etc. Note that `Cache-Control` header is written by Next.js automatically when using static site generation (see https://github.com/vercel/next.js/issues/22319), there is a new fix currently implemented that allows a custom setting, but an additional middleware or proxy could be needed to send the customized Cache-Control headers to the CDN.

As I'm currently migrating my site to Faust.js, I'll share my experiences with this as soon I got further with this
